### PR TITLE
More ENV rdoc

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: hash.c [hunk 1]
```diff
@@ -5727,12 +5727,16 @@ env_keep_if(VALUE ehash)
 }
 
 /*
- *  call-seq:
- *     ENV.slice(*keys) -> a_hash
- *
- *  Returns a hash containing only the given keys from ENV and their values.
+ * call-seq:
+ *   ENV.slice(*names) -> Hash
```
Comment (nobu) [inline hash.c]: The return value is not `Hash` itself.

Comment (BurdetteLamar) [reply hash.c]: There are many other such, including Hash, Array, Integer, Enumerator, etc.  The call-seq returns are among the least consistent things here.   They probably should be revised wholesale, after a discussion of how they should be.

Meanwhile, what would you like this one to say?

File: hash.c [hunk 2]
```diff
@@ -6035,7 +6039,7 @@ env_has_value(VALUE dmy, VALUE obj)
 
 /*
  * call-seq:
- *   ENV.rassoc(value)
+ *   ENV.rassoc(value) -> [value, name] or nil
```
Comment (nobu) [inline hash.c]: The order is reverse.

File: hash.c [hunk 3]
```diff
@@ -6166,11 +6170,21 @@ env_f_to_hash(VALUE _)
...
- *
- * Creates a hash with a copy of the environment variables.
+ *   ENV.to_h                        -> hash of name/value pairs
+ *   ENV.to_h {|name, value| block } -> hash of name/value pairs
  *
+ * With no block, returns a Hash containing all name/value pairs from ENV:
+ *   ENV.replace('foo' => '0', 'bar' => '1')
+ *   ENV.to_h # => {"bar"=>"1", "foo"=>"0"}
+ * With a block, returns a Hash whose items are determined by the block.
+ * Each name/value pair in ENV is yielded to the block as a 2-element Array.
```
Comment (nobu) [inline hash.c]: Two arguments are yielded, not an `Array`.

File: hash.c (hunk 2 above)
Comment (BurdetteLamar) [reply hash.c]: Right you are.  (Reversed too much!)  Fixing.

File: hash.c [hunk 4]
```diff
@@ -6323,15 +6355,38 @@ env_update_block_i(VALUE key, VALUE val, VALUE _)
...
+ * Deletes the ENV entry for a hash value that is +nil+:
+ *   ENV.merge!('baz' => nil, 'bat' => nil) # => {"bar"=>"1", "foo"=>"0"}
+ * For an already-existing name, if no block given, overwrites the ENV value:
+ *   ENV.merge!('foo' => '4') # => {"bar"=>"1", "foo"=>"4"}
+ * For an already-existing name, if block given,
+ * yields the name, its ENV value, and its hash value;
+ * the  block's return value becomes the new name:
+ *   ENV.merge!('foo' => '5') { |name, env_val, hash_val | env_val + hash_val } # => {"bar"=>"1", "foo"=>"45"}
+ * Raises an exception if a name or value is invalid
+ * (see {Invalid Names and Values}[#class-ENV-label-Invalid-Names+and+Values]);
```
Comment (MSP-Greg) [inline hash.c]: Several Errors existed in the file with the linking to the Class 'Introduction' headings.

All the id's are of the form:
```
#class-ENV-label-<heading text>
```
With spaces in \<heading text\> replaced by plus signs.

So the link should be:
```
[#class-ENV-label-Invalid+Names+and+Values
```

Comment (BurdetteLamar) [reply hash.c]: Oh, wow!  Thanks MSP-Greg.  Correcting now.

Comment (MSP-Greg) [reply hash.c]: Thanks for working on docs.  Sorry for the above, as I didn't mention that I was referring to issues with RDoc, etc.  You made the connection...

I noticed it with:

https://msp-greg.github.io/ruby_trunk/Core/ENV.html

then checked

https://docs.ruby-lang.org/en/master/ENV.html

Comment (BurdetteLamar) [reply hash.c]: I think it's fixed now.

**Pull request merged**